### PR TITLE
feat: run shellcheck & add github action

### PR DESCRIPTION
**Pull request by pimlie**
This PR cleans up the bash scripts by fixing all issues except SC2001 reported by shellcheck. The SC2001 issue is an info level message which I think would be nice to clean up, but left that in for now.

Also added a github action to run shellcheck to ensure there are no issues in future prs

### CODE REVIEW

**Comment by berglh** [comment]
Hi @pimlie,

Thanks for the PR, sorry I seem to have missed it in the noise. This seems well worth doing and appreciate the time you put in to check the scripts for safety with shellcheck.

I'll try to pull the branch in locally and test it out from start to finish.

The main thing I was concerned about looking at your PR was double-double quotes the with the variable assignment with default value fallback after the read prompt; testing this manually in bash seems to work. I've heard of shellcheck before, and I guess we should trust the recommendations from the experts :)

File: sbin/mok-setup.sh [hunk 1]
```diff
@@ -17,12 +17,12 @@ else
...
+SCRIPT_DIR="$( cd "$( dirname "$(readlink -f "${BASH_SOURCE[0]}")" )" &> /dev/null && pwd )"
 REQURIED_PACKAGES=("mokutil" "openssl" "sbsigntool" "grub-efi-amd64-signed" "fwts")
 
-for dpkg in ${REQURIED_PACKAGES[@]}; do
-    printf "Checking if ${dpkg} is installed..";
-    dpkg -s ${dpkg} | grep Status | grep -q installed;
-    if [ "$?" != 0 ]; then
+for dpkg in "${REQURIED_PACKAGES[@]}"; do
+    echo -n "Checking if ${dpkg} is installed..";
+   if ! dpkg -s "${dpkg}" | grep Status | grep -q installed; then
```
Comment (berglh) [inline sbin/mok-setup.sh]: @pimlie The indentation here on the if statement needs another space if I'm reading this correctly.

**Review by berglh — COMMENTED** [review]
I only saw the one indent issue.

**Comment by pimlie** [comment]
Thanks for the review, have fixed the indentation. And yeah, some of the shellcheck recommendations might look 'off' at first. Had the exact same concerns when I first ran shellcheck :)

Thanks for sharing these scripts btw, as fyi I'm referring to your repository/scripts here: https://github.com/pimlie/ubuntu-mainline-kernel.sh#secureboot

**Comment by berglh** [comment]
This all LGTM. Thanks for fixing the linting.

I'm new to the GitHub workflows, in GitLab we need to enable CI on the repository for the functionality to work. Do I need to do the same here or will just including the file enable it?

**Pull request merged**